### ISSUE TITLE
Cannot end Syncbase Java server or cancel watch

### ISSUE THREAD

**Issue by AsturaPhoenix** [vanadium/issues#809]
Since Syncbase holds a lock on its backing file db, it forces us to do proper lifecycle management, which is good to do anyway. However, when I use a `watch`, I cannot seem to terminate it or the Syncbase server.

Things tried:
in Android service `onDestroy`:
`V.shutdown()` - returns immediately, but restarting Syncbase reports file lock
`server.stop()` - never returns; attempting to execute on another thread does not release lock
`server.stop()`, with `Stream<WatchChange>.cancel` in client - `cancel` never returns (did not try calling `server.stop` while `cancel` attempt was in progress)

It is worth noting that while I was calling cancel, another thread was waiting for a change from the `Stream<WatchChange>`. Which, incidentally, the `Stream` is an `Iterable`, but as a developer I can't help but doubt that it follows Iterable semantics wrt. `iterator()` calls - do subsequent `iterator()` calls start watching from the resume marker? Do parallel iterators work right? It would help for peace of mind if this were spelled out in the Javadoc.

Without `watch`, `server.stop()` returns quickly, but restarting fails on the same file lock.

**Comment by AsturaPhoenix** [comment]
James's recommended workaround of System.exit() in lieu of Syncbase/V23 shutdown works.

**Comment by kenashcraft** [comment]
In all of my testing of syncslides, I've never seen this file lock issue.
Hmmm...what are you doing differently?  Are you running syncbase in a
Service?

**Comment by AsturaPhoenix** [comment]
Yep. It's a bound service, and lifecycle callbacks are being called to indicate destruction, but the Syncbase go server stays alive, certainly when watch is used. I don't know why I still have the lock issue without watch; it's possible running it as an Android service, even bound, somehow prevents the JVM sandbox from terminating without an explicit call.

**Comment by AsturaPhoenix** [comment]
To update, I'm cancelling the watch via the context now and Server.stop is returning, but the file lock issue is still occuring and it seems to be separate. Stream.cancel will soon no longer be an option with an upcoming change from @spetrovic77.

**Comment by AsturaPhoenix** [comment]
So it looks like the reason Syncslides never ran into this problem is that it keeps its Syncbase instance in a singleton which has the same lifecycle as the JVM. Modelling it as a service that could restart within the same process (even with V.shutdown/V.init boundaries) seems to trigger this issue.

**Comment by AsturaPhoenix** [comment]
This makes it hard to segment integration tests.
